File: scala/stubs.scala
import org.apache.spark.mllib.linalg.Vector
import org.apache.spark.rdd.RDD
import org.apache.spark.{SparkConf, SparkContext}
import org.apache.spark.mllib.clustering.{KMeansModel, KMeans}

/**
 * 
 */
def clustersInfo(model: KMeansModel) = {
  val centroids = model.clusterCenters
  centroids.foreach(l => println(s"Class: ${l(0).toInt}, Age: ${l(1).toInt}, Fair: ${l(2).toInt}"))
}


def featureEngineering(data : RDD[String]): RDD[Vector] = {

	data.map(line => {

	  val values: Array[String] = line.split('|')

	  // TODO: transform 'values' which is an array of String to an array of Double to be read by MLlib

	  val numericalData: Array[Double] = ???

	  Vectors.dense(numericalData)
	})

}

// Loading data
// TODO: read file ./src/main/resources/data_titanic.csv

// Feature Engineering
// TODO: use the featureEngineering method to get the cleaned data.
	
// Modelling
// TODO: Train a KMeans model on the data set

// Inspect centroid of each cluster
println("Clusters description")
// TODO: For each cluster, print the centroid information. You can use the clustersInfo method in

File: python/stubs.py
import numpy as np
import scipy.sparse as sps
from pyspark.mllib.linalg import Vectors
from pyspark.mllib.clustering import KMeans, KMeansModel

def clustersInfo(model):
	centroids = model.clusterCenters
	for c in centroids:
		print "Class: ", int(c[0]), ", Age: ", int(c[1]), ", Fair: ", int(c[2])

def transformLine(line):
	values = line.split('|')

	# TODO: transform 'values' which is an array of String to an array of Double to be read by MLlib
	# Carefull: you need to deal with missing data (in this dataset all missing data are reprsented by 'NA')
	numericalData = ???
	return Vectors.dense(numericalData)


def featureEngineering(data):
	return data.map(transformLine)


# Loading data
# TODO: read file ./src/main/resources/data_titanic.csv

# Feature Engineering
# TODO: use the featureEngineering method to get the cleaned data.
	
# Modelling
# TODO: Train a KMeans model on the data set

# Inspect centroid of each cluster
print "Clusters description"
# TODO: For each cluster, print the centroid information. You can use the clustersInfo method in


